src/zkp/pisch.rs

```diff
@@ -107,6 +102,12 @@ impl<'a> Debug for ProverSecret<'a> {
     }
 }
 
+impl AsRef<utils::CurvePoint> for utils::CurvePoint {
+    fn as_ref(&self) -> &utils::CurvePoint {
+        self
+    }
+}
```

Comment:
todo: This should live next to the definition of the `CurvePoint` type in `utils.rs`.